dvc/repo/experiments/executor/base.py

```diff
@@ -450,9 +478,21 @@ def _repro_args(cls, dvc):
             kwargs = {}
         return args, kwargs
 
+    @staticmethod
+    def _auto_push(dvc: "Repo", scm: "Git"):
+        git_remote = os.getenv(DVC_EXP_GIT_REMOTE)
```

Comment:
It's not really a requirement for this PR, but one thing to consider would be designing this to be more like

```py
def _exp_push(dvc, scm, git_remote, push_cache=True, run_cache=True):
    ...
    dvc.experiments.push(
        git_remote, branch, push_cache=push_cache, run_cache=run_cache
    )
```

It seems like eventually we are going to end up needing to provide some API that supports finer control over this pushing behavior, and triggering it won't always be based on env vars. So the automated push functionality should be designed with that in mind.

Also with the current implementation we end up re-reading environment vars over and over again. It would be cleaner to just read them all once at the start of `executor.reproduce()` and then pass the known values into functions where they are needed, with something like:

```py
    def reproduce(...):
        auto_push = env2bool(DVC_EXP_AUTO_PUSH)
        git_remote = os.getenv(DVC_EXP_GIT_REMOTE)
        ...
        if auto_push:
            _exp_push(..., git_remote, ...)
```